src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/TypedArray.cs

```diff
@@ -66,40 +66,15 @@ protected TypedArray(ArrayBuffer buffer, int byteOffset) : base(typeof(T).Name,
         protected TypedArray(ArrayBuffer buffer, int byteOffset, int length) : base(typeof(T).Name, buffer, byteOffset, length)
         { }
 
-        protected TypedArray(SharedArrayBuffer buffer) : base(typeof(T).Name, buffer)
-        { }
-
-        protected TypedArray(SharedArrayBuffer buffer, int byteOffset) : base(typeof(T).Name, buffer, byteOffset)
-        { }
-
-        protected TypedArray(SharedArrayBuffer buffer, int byteOffset, int length) : base(typeof(T).Name, buffer, byteOffset, length)
-        { }
-
         internal TypedArray(IntPtr jsHandle) : base(jsHandle)
         { }
 
         public TypedArrayTypeCode GetTypedArrayType()
         {
             switch (this)
             {
-                case Int8Array _:
-                    return TypedArrayTypeCode.Int8Array;
                 case Uint8Array _:
```

Comment:
If we're going to keep this type generic, we might as well keep the support for the various primitive element types. If we're only keeping Uint8Array we should just rename it and remove the generic parameter entirely.

Reply:
To keep the code simplification, we can just delete all the specific derived types and keep the base class (making it no longer abstract), and check T at construction time to make sure it's valid. We already have GetTypedArrayType, so we can use that to perform the check.

Reply:
makes sense, thanks! I will remove `TypedArray`